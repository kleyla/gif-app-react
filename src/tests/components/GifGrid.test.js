const { shallow } = require("enzyme");
const { GifGrid } = require("../../components/GifGrid");
const { useFetchGifs } = require("../../hooks/useFetchGifs");
jest.mock("../../hooks/useFetchGifs");

describe("Prueba de GifGrid", () => {
  const category = "pika";

  test("Debe mostrarse correctamente", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe demostrar items cuando se cargan imagenes useFetchGifs", () => {
    const gifs = [
      { id: "abc", url: "https://", title: "pika" },
      { id: "123", url: "https://holi", title: "kiss" },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });
});
