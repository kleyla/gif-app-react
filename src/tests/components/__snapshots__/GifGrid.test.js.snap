// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de GifGrid Debe demostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    pika
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="pika"
      url="https://"
    />
    <GifGridItem
      id="123"
      key="123"
      title="kiss"
      url="https://holi"
    />
  </div>
</Fragment>
`;

exports[`Prueba de GifGrid Debe mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    pika
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
